From 805527a087fa67d13c2aea3051944423faae2833 Mon Sep 17 00:00:00 2001
From: Han Xu <b45815@freescale.com>
Date: Fri, 27 Mar 2015 11:23:08 -0500
Subject: [PATCH 0738/5242] MLK-10497 ARM: dts: supported spi-nor on imx6qp
 sabreauto

commit  062408d56557a247f3d85217d86dd848aff37423 from
https://source.codeaurora.org/external/imx/linux-imx.git

add new dts to support spi-nor on imx6qp sabreauto

Signed-off-by: Han Xu <b45815@freescale.com>
(cherry picked and merge from commit edb500d09c08effce28f8c45490372f38b037815)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/boot/dts/Makefile                   |    1 +
 arch/arm/boot/dts/imx6qp-sabreauto-ecspi.dts |   43 ++++++++++++++++++++++++++
 2 files changed, 44 insertions(+)
 create mode 100644 arch/arm/boot/dts/imx6qp-sabreauto-ecspi.dts

diff --git a/arch/arm/boot/dts/Makefile b/arch/arm/boot/dts/Makefile
index 00378dd..5eec57c 100644
--- a/arch/arm/boot/dts/Makefile
+++ b/arch/arm/boot/dts/Makefile
@@ -523,6 +523,7 @@ dtb-$(CONFIG_SOC_IMX6Q) += \
 	imx6qp-nitrogen6_som2.dtb \
 	imx6qp-phytec-mira-rdk-nand.dtb \
 	imx6qp-sabreauto.dtb \
+	imx6qp-sabreauto-ecspi.dtb \
 	imx6qp-sabreauto-gpmi-weim.dtb \
 	imx6qp-sabresd.dtb \
 	imx6qp-sabresd-btwifi.dtb \
diff --git a/arch/arm/boot/dts/imx6qp-sabreauto-ecspi.dts b/arch/arm/boot/dts/imx6qp-sabreauto-ecspi.dts
new file mode 100644
index 0000000..ace6da9
--- /dev/null
+++ b/arch/arm/boot/dts/imx6qp-sabreauto-ecspi.dts
@@ -0,0 +1,43 @@
+/*
+ * Copyright (C) 2015 Freescale Semiconductor, Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2 as
+ * published by the Free Software Foundation.
+ */
+
+#include "imx6qp-sabreauto.dts"
+
+&ecspi1 {
+	pinctrl-assert-gpios = <&gpio5 4 GPIO_ACTIVE_LOW>;
+	status = "okay";
+};
+
+/* remove #if 0 after flexcan is ready */
+#if 0
+&flexcan2 {
+	/* max7310_c on i2c3 is gone */
+	status = "disabled";
+};
+#endif
+
+&i2c3 {
+	/* pin conflict with ecspi1 */
+	status = "disabled";
+};
+
+&uart3 {
+	/* the uart3 depends on the i2c3, so disable it too. */
+	status = "disabled";
+};
+
+&usbh1 {
+	/* max7310_b on i2c3 is gone */
+	status = "disabled";
+};
+
+&usbotg {
+	/* max7310_c on i2c3 is gone */
+	status = "okay";
+	dr_mode = "peripheral";
+};
-- 
1.7.9.5

